test/integration/tasks.test.js

```diff
@@ -138,6 +138,46 @@ describe("Tasks", function () {
           return done();
         });
     });
+    it("Should get all the list of tasks by paginated", function (done) {
+      const limit = 1;
+      const chaiRequester = chai.request(app).keepOpen();
+      chaiRequester
+        .get("/tasks")
+        .query({ limit })
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.tasks).to.be.a("array");
+          expect(res.body.tasks.length).to.equal(limit);
+          const offset = { ...res.body.tasks[0] };
+
+          chaiRequester
+            .get("/tasks")
+            .query({ offset: offset.id })
+            .end((err, res) => {
+              if (err) {
+                return done(err);
+              }
+              expect(res).to.have.status(200);
+              expect(res.body).to.be.a("object");
+              expect(res.body.tasks).to.be.a("array");
+              const taskWithParticipants = res.body.tasks;
+              taskWithParticipants.forEach((task) => {
+                if (task.type === "group") {
+                  expect(task.participants).to.include(appOwner.username);
+                } else {
+                  expect(task.assignee).to.equal(appOwner.username);
+                }
```

Comment:
I wonder if this responsibility to validate the correct type of task should ideally belong to a test that checks for types. Not to the test that checks for pagination.

Reply:
Noted